### PR TITLE
W-12625688: Avoid errors caused by using mock/spy to ClassLoader

### PR DESCRIPTION
**Pull request by hyerans**

### CODE REVIEW

**Comment by eze210** [comment]
--validate

**Comment by eze210** [comment]
There is a test (at least) that failed for me in spring-config module for the same reason (`LazyComponentInitializerAdapterTestCase`). Can you add the dependency there too?
I added it in another PR (#12420), but I think it makes more sense to add it here.

**Comment by hyerans** [comment]
> There is a test (at least) that failed for me in spring-config module for the same reason (`LazyComponentInitializerAdapterTestCase`). Can you add the dependency there too? I added it in another PR (#12420), but I think it makes more sense to add it here.

Working on it now. Thanks Eze!

**Comment by hyerans** [comment]
--validate

File: modules/extensions-support/pom.xml [hunk 1]
```diff
@@ -249,6 +249,12 @@
             <version>${project.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-inline</artifactId>
+            <version>4.11.0</version>
```
Comment (elrodro83) [inline modules/extensions-support/pom.xml:255]: use the `${mockitoVersion}` property

File: modules/launcher/pom.xml [hunk 2]
```diff
@@ -77,5 +77,11 @@
             <scope>test</scope>
             <type>test-jar</type>
         </dependency>
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-inline</artifactId>
+            <version>4.11.0</version>
```
Comment (elrodro83) [inline modules/launcher/pom.xml:83]: use the `${mockitoVersion}` property

**Review by elrodro83 — CHANGES_REQUESTED** [review]

**Review by elrodro83 — APPROVED** [review]

**Pull request merged**